kafka-client/src/main/java/dev/responsive/kafka/api/async/internals/AsyncProcessor.java

```diff
@@ -69,12 +66,18 @@
  * -Coordinates the handoff of records between the StreamThread and AyncThreads
  * -The starting and ending point of all async events -- see {@link AsyncEvent}
  */
-public class AsyncProcessor<KIn, VIn, KOut, VOut>
-    implements Processor<KIn, VIn, KOut, VOut>, FixedKeyProcessor<KIn, VIn, VOut> {
+public class AsyncProcessor<KIn, VIn, KOut, VOut> {
```

Comment:
Since this no longer implements the `Processor` interfaces, I think I'll rename it to "AsyncProcessingEngine" or something to that effect (suggestions welcome). And then rename the new wrapper class that creates/delegates to this class to "AsyncProcessor". But I'll wait until the PR has been reviewed/approved to do the renamings since that might impact a bunch of code